core/services/relay/evm/evm.go

```diff
@@ -209,7 +226,24 @@ func (r *Relayer) NewConfigProvider(args commontypes.RelayArgs) (commontypes.Con
 		return nil, fmt.Errorf("internal error: chain id in spec does not match this relayer's chain: have %s expected %s", relayConfig.ChainID.String(), r.chain.ID().String())
 	}
 
-	configProvider, err := newConfigProvider(lggr, r.chain, relayOpts)
+	// Handle legacy jobs which did not yet specify provider type and
+	// switched between median/mercury based on presence of feed ID
+	if args.ProviderType == "" {
+		if relayConfig.FeedID == nil {
+			args.ProviderType = "median"
+		} else {
+			args.ProviderType = "mercury"
+		}
+	}
+
+	switch args.ProviderType {
+	case "median":
+		configProvider, err = newStandardConfigProvider(lggr, r.chain, relayOpts)
```

Comment:
Is this really only for median? Or is it for 'not mercury'?

Reply:
This is only ever called by the bootstrapper so it is always either `median` or `mercury`. LLO PR will add a new type `llo` that needs a third type of config contract.

We could maybe rename to `standard`/